feat: add native wsteth support

tx-nikola opened the pull request:
- Tested withdraw, deposit, and transfer for the native wstETH token, as well as the bridged (erc20) one, also tested ETH token and others to not mess up the previous flow.

- The custom gas price fetching and custom withdraw tx creation had to be added since newer version of the SDK works in a different way for bridge addresses and it didn't work with native wstETH address

- The balances/tokens map had to be updated - previously it used L1 address as the key, but I had to change it so that it uses the L2 address as the key - since we want to keep backward compatibility with the previous bridged (non-native non-Lido) wstETH token, so we want to support both, but they have the same L1 address, so we couldn't use the L1 address as the key

This PR fixes https://github.com/matter-labs/dapp-portal/issues/238

JackHamer09 commented on composables/zksync/deposit/useTransaction.ts:
```diff
@@ -1,23 +1,131 @@
   const eraWalletStore = useZkSyncWalletStore();
+  const { eraNetwork } = useZkSyncProviderStore();
   const { captureException } = useSentryLogger();
+  const chain = eraNetwork.key === "mainnet" ? mainnet : sepolia;
```
this doesn't make a lot of sense to me.. Why not use l1 network config of selected network?

JackHamer09 commented on composables/zksync/deposit/useTransaction.ts:
```diff
@@ -1,23 +1,131 @@
+    fee: DepositFeeValues
+  ) => {
+    // Create signer
+    const { ethereum } = window as any;
```
This would only support injected wallets... You should use connected wallet instead of creating new one

JackHamer09 commented on composables/zksync/deposit/useTransaction.ts:
```diff
@@ -1,23 +1,131 @@
+  ) => {
+    // Create signer
+    const { ethereum } = window as any;
+    const browserProvider = new BrowserProvider(ethereum);
+    const isMainnet = eraNetwork.key === "mainnet" || eraNetwork.id === 324;
+    const networkType = isMainnet ? types.Network.Mainnet : types.Network.Sepolia;
```
this looks like AI generated code which doesn't make too much sense

JackHamer09 commented on composables/zksync/deposit/useTransaction.ts:
```diff
@@ -1,23 +1,131 @@
+    const allowance = await signer.getAllowanceL1(transaction.tokenAddress, transaction.bridgeAddress);
+
+    if (allowance < BigInt(transaction.amount)) {
+      const approveTx = await signer.approveERC20(transaction.tokenAddress, transaction.amount, {
```
Allowance should be done properly and handled on the UI, there is already logic for it

JackHamer09 reviewed (changes requested):
tldr: code is just ai generated and the quality is really poor

JackHamer09 reviewed (changes requested):
1. Please check this PR with some improvements https://github.com/matter-labs/dapp-portal/pull/253
2. See the comments, left some feedback about general logic of token addresses
3. I'm getting a bunch of errors in `views/transactions/Deposit.vue`. Mostly `Property 'l2Address' does not exist on type 'Token'.`. I tried rebuilding everything and installing latest dependencies.

4.
> I left the allowance part - the reason I added it for the handleCustomBridgeDeposit function is because before, when calling wallet.deposit, we pass the approveBaseERC20 as true

It was a hacky way of fixing deposit for those chains that use custom erc20 as base token I think. Normally, and in this case as well, you should do it correctly through the UI, so that user understands what's going on and which transaction they are signing. Or is this already handled?

tx-nikola commented:
@JackHamer09 

- Removed all occurencies of this pattern:
```
{
<object properties>,
...(transaction.value.token.l2BridgeAddress ? { bridgeAddress: transaction.value.token.l2BridgeAddress } : {}),
}
```
I used it because in some cases in other projects, I found it that some functions that expect object params, expect either an object with a parameter or with no parameter, and even if we pass "undefined" as a param, the function will take it as a param that exists, and potentially throw an error - so I just adopted the style that if I pass the object with potentially undefined params, I don't pass the params that would be undefined. In these cases it doesn't make a difference, so I agree with you to keep it simplified.

- I updated changes for tokens.ts to not be hacky
  - Instead of mapping L1 and L2 token objects in a complicated way, I created new mapper functions in `helpers.ts` and use them wherever we need balances/tokens arrays.
  Before, it was hacky because I was trying to change the object mapping - it was `{[l1Address]: Token}` because the object keys must be unique and there can't be 2 same L1 addresses (as we have in the case of wstETH) but now I left it as it is and just map it whenever we use the array.
